superduperdb/backends/sqlalchemy/metadata.py

```diff
@@ -352,29 +350,37 @@ def update_metadata(self, key, value):
     def add_query(self, query: 'Select', model: str):
         query_serialized = query.serialize()
         query_hash = str(hash(json.dumps(query_serialized)))
+
         with self.session_context() as session:
             with self._lock:
                 row = {
-                    'query_id': self._query_watermark,
                     'query': query_serialized,
                     'model': model,
-                    'hash': query_hash,
+                    'query_id': query_hash,
```

Comment:
`query_id` was incremented in memory - would only work within session. Replacing this integer-id with 
the hash as the id. This means that queries automatically carry their id around with them.